wolfssl/internal.h

```diff
@@ -4489,6 +4502,19 @@ struct WOLFSSL {
 #endif
 };
 
+/*
+ * The SSL object may have its own certificate store. The below macros simplify
+ * logic for choosing which WOLFSSL_CERT_MANAGER and WOLFSSL_X509_STORE to use.
+ * Always use SSL specific objects when available and revert to CTX otherwise.
+ */
+#ifdef WOLFSSL_LOCAL_X509_STORE
+#define SSL_CM(ssl) (ssl->x509_store_pt ? ssl->x509_store_pt->cm : ssl->ctx->cm)
+#define SSL_STORE(ssl) (ssl->x509_store_pt ? ssl->x509_store_pt : \
+                  (ssl->ctx->x509_store_pt ? ssl->ctx->x509_store_pt : \
+                                            &ssl->ctx->x509_store))
+#else
+#define SSL_CM(ssl) ssl->ctx->cm
```

Comment:
That's what I thought as well but `x509_store_pt` is only defined with `WOLFSSL_LOCAL_X509_STORE` so there is no point in defining `SSL_STORE` in this case.

Reply:
I searched a saw a place `SSL_STORE ` could be used. It is still a good idea for future protections too. Please add a `SSL_STORE` macro for the else case.

Reply:
Do you mean the `LoadCertByIssuer` call in `ProcessPeerCerts`? Also changed in `wolfSSL_set0_verify_cert_store` and `wolfSSL_set1_verify_cert_store`.

I still don't see why we should define `SSL_STORE` in the `#else` case. When `WOLFSSL_LOCAL_X509_STORE` is not defined then no `x509_store*` members are defined in `WOLFSSL` or `WOLFSSL_CTX`. I think its safer and easier to debug to have wolfSSL error out during compilation than during runtime (in the case of just `#define SSL_STORE(ssl) NULL`). Especially since this is an internal macro.